python/mdvtools/tests/test_chunked_normalization.py

```
@@ -0,0 +1,227 @@
+        print(f"Dense 100k x 10k matrix: {dense_memory:.1f}MB")
+        assert dense_memory > 0
+        assert dense_memory > sparse_memory  # Dense should use more memory
+    
+    def test_chunked_vs_traditional_performance(self):
+        """Compare performance of chunked vs traditional normalization."""
+        # Create moderately large matrix
+        n_cells, n_genes = 50000, 5000
+        sparsity = 0.95
+        nnz = int(n_cells * n_genes * (1 - sparsity))
+        
+        print(f"\nComparing chunked vs traditional normalization")
```

Flake8 (7.2.0)

[error] f-string is missing placeholders

(F541)